examples/registration_form/lib/root/root_component.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:angular/angular.dart';
+import 'package:angular_components/angular_components.dart';
+import 'package:angular_forms/angular_forms.dart';
+import 'package:examples.registration_form/address/address_component.dart';
+
+@Component(
+    selector: 'hello-world',
+    templateUrl: 'root_component.html',
+    directives: [
+      formDirectives,
+      materialInputDirectives,
+      MaterialYesNoButtonsComponent,
+      MaterialSaveCancelButtonsDirective,
+      NgIf,
+      AddressComponent
+    ],
+    providers: [
+      popupBindings,
+    ])
+class RootComponent {
+  @ViewChild('myForm')
+  ControlContainer myForm;
+
+  String firstName;
+  String lastName;
+
+  Person person;
+
+  void onSubmit() {
+    person = _createPerson();
+  }
+
+  // TODO(alorenzen): Reset form inputs.
+  void onClear() {
+    person = null;
+  }
+
+  Person _createPerson() {
+    return new Person(
```

Comment:
Since you are only using the values I think it is cleaner to do:
Map<String, dynamic> values = form.value;
firstName: values['first-name']

Also some people like having const strings for the values. Instead of using the string everywhere. I personally don't think it is needed, but wanted to point it out if you liked that pattern better.

Reply:
Ah, I like this much more than what I had originally.